### ISSUE TITLE
[NV-474] - Fit to screen's lentgh/size

### ISSUE THREAD

**Issue by novubot** [novuhq/novu#666]
## Why?

For some screens the bottom third is totally empty today. For larger screen it's very limiting the usable space.

## How?

Let's take advantage of that space by modifying the UI to fit the screen size. Making the screen fit inside the viewable port 100vh-(the existing header and spacing)

### Where?

In the apps/web project, to get the latest workflow editor, checkout from the `workflow` branch.

See:

![image.png](https://uploads.linear.app/f13f1996-c9b0-4fea-8ee7-2c3faf6a832d/6ee79268-bc4c-472e-8c95-e46c0c415524/8db5fc12-58ae-4334-81ba-2dbd2d75af70)

\--\
Created via [Raycast](https://www.raycast.com?ref=signatureLinear)

**Comment by AakashRaj20** [comment]
@scopsy hey I would like to try this issue

**Comment by fosa-ij** [comment]
> Hi @fosa-ij sure! So you would need some understanding of React and CSS for this to work. In the screenshot, You can see that the card of the "Workflow Editor" only takes 60-70 percent of the screen height. What we won't is to utilize the entire screen height of the user.
> 
> If this still doesn't make sense feel free to reach out on Discord, and if you think you want to try something else, let me know so I can assign @AakashRaj20 for it 🙏

okay @scopsy thanks a lot, but React is not part of my stack, I would like to pass it on to @AakashRaj20.
But anything in the mix of CSS and javascript would fit just fine. Thanks for the effort.

**Comment by davidsoderberg** [comment]
@AakashRaj20 I have now assigned you 😄

**Comment by AakashRaj20** [comment]
> @AakashRaj20 I have now assigned you smile

Thank you 😁

**Comment by AakashRaj20** [comment]
> @AakashRaj20 I have now assigned you smile

Hey I have tried setting up the project multiple times but nothing loads on the web page it remains empty. Please help me setting up the project Locally.

**Comment by scopsy** [comment]
Sure @AakashRaj20 could you please send me a DM on Discord so we can schedule a call ?

**Comment by scopsy** [comment]
@AakashRaj20 unfortuantly looks like we couldn't meet, let me know if you want to try and schedule again 🙏

**Comment by AakashRaj20** [comment]
> @AakashRaj20 unfortuantly looks like we couldn't meet, let me know if you want to try and schedule again 🙏
Yes I tried setting up the project but I was not  able to it would he great if you can take a look again tommorow any time you are free.

**Comment by scopsy** [comment]
Couldn't find you on Discord @AakashRaj20 what is your handle there?

**Comment by AakashRaj20** [comment]
> Couldn't find you on Discord @AakashRaj20 what is your handle there? 

itachi20#7614

**Comment by jainpawan21** [comment]
Hi @AakashRaj20 
Do you need any help in this?
You can message us in help-needed channel in discord or DM me

**Comment by Ian-Yy** [comment]
Hi, can I give a try on this issue?

**Comment by Ian-Yy** [comment]
Hi, I have analyzed the issue and would like some help on this.

I think this is happening because the `PageContainer` does not fit to screen size too. Thus, even setting the React Flow in the Workflow Editor `height: 100%` would not work. I tried setting both the `PageContainer` and React Flow to 100% and it works but it will affect every other pages. Another method is to hardcode the height of React Flow to 60-70vh.

Please let me know which methods will be better or if there's a better way. Thanks

**Comment by scopsy** [comment]
@Ian-Yy do you have a screen shot of how it will affect other pages?

**Comment by Ian-Yy** [comment]
Sorry for taking so long on this. When I was taking the screen shot for the other pages, I realized setting the `height: 100%` will break those pages where their height needs to be more than 100% such as `Getting Started`, `Integration Store` and etc. So, I personally think to minimize the affected page, it would be better setting the height of React Flow to vh. 

I also tried setting top down `minHeight: 100%` which should make every page utilize 100% of the height but I have no idea why it would not work on the novu. I even have a working code snippet that is structurally similar that works. Just let me know if you think setting all pages to utilize 100% height is a better direction then I can get the code snippet up and troubleshoot why it does not work on novu.
